src/ui/App/App.tsx

```diff
@@ -433,28 +446,28 @@ export interface AppProps {
 }
 
 export function App({ initialView, mode, inspect }: AppProps) {
+  const windowContext = WindowContext.get();
   const bodyClassList = useMemo(() => {
     const result = [];
-    if (templateData.windowContext === 'dialog') {
+    if (windowContext.isDialog()) {
       result.push(styles.isDialog);
-    } else if (templateData.windowContext === 'tab') {
+    } else if (windowContext.isTab()) {
       result.push(styles.isTab);
     }
-    if (mode === 'onboarding' || templateData.layout === 'page') {
+    if (mode === 'onboarding' || windowContext.hasPageLayout()) {
       result.push(styles.pageLayout);
     }
     return result;
-  }, [mode]);
+  }, [mode, windowContext]);
 
   const { connected } = useStore(runtimeStore);
 
   useBodyStyle(
     useMemo(() => ({ opacity: connected ? '' : '0.6' }), [connected])
   );
 
-  const isOnboardingTemplate =
+  const isOnboardingView =
     mode === 'onboarding' && initialView !== 'handshakeFailure';
```

Comment:
I'm not sure what the "template" suffix is supposed to mean here, but I find it easier to think of it as "onboarding view